Encryption.md

```diff
@@ -132,49 +203,164 @@ the encryption buffer is comprised of an IV, ciphertext and tag, as described in
 Algorithms section. The length of the encryption buffer (a 4-byte little endian) is 
 written to the output stream, followed by the buffer itself.
```

Comment:
This statement disagrees with the length definition below. This states that the encryption buffer is the IV, ciphertext and tag and that the length written is the length of the encryption buffer. The diagram states that the ciphertext is "`length` bytes".

My comment about moving the length assumes that the length is for the ciphertext, not the whole encryption buffer. Length should be placed directly before the bytes that it applies to. So it's current position is fine if it is the length of the encryption buffer. If it is the length of the ciphertext, then it should be just before the ciphertext.